Content/Items/Misc/Weapons.Earthduster.cs

```diff
@@ -0,0 +1,560 @@
+
+            Projectile.position = armPos - Projectile.Size * 0.5f;
+
+            Projectile.spriteDirection = Projectile.direction;
+
+            if (Main.myPlayer == Projectile.owner && !reloading)
+            {
+                float interpolant = Utils.GetLerpValue(1f, 5f, Projectile.Distance(Main.MouseWorld), true);
+
+                Vector2 oldVelocity = Projectile.velocity;
+
+                Projectile.velocity = Vector2.One.RotatedBy(Vector2.Lerp(Projectile.velocity, owner.DirectionTo(Main.MouseWorld), interpolant).ToRotation() - MathHelper.PiOver4);
+                if (Projectile.velocity != oldVelocity)
+                {
+                    Projectile.netSpam = 0;
+                    Projectile.netUpdate = true;
+                }
+            }
+        }
+
+        public override bool PreDraw(ref Color lightColor)
+        {
+            if (!draw)
+                return false;
+
+            Texture2D tex = ModContent.Request<Texture2D>(Texture).Value;
+            Texture2D dirtTex = ModContent.Request<Texture2D>(Texture + "_Dirt").Value;
+            Texture2D glowTex = ModContent.Request<Texture2D>(Texture + "_Glow").Value;
+            Texture2D bloomTex = ModContent.Request<Texture2D>(AssetDirectory.Keys + "GlowAlpha").Value;
+
+            Vector2 offset = Vector2.Lerp(Vector2.Zero, Vector2.UnitY.RotatedBy(Projectile.rotation + (owner.direction == -1 ? MathHelper.Pi : 0f)) * 13f, shots / (float)MAXSHOTS);
+            Main.spriteBatch.Draw(dirtTex, (Projectile.Center + offset) - Main.screenPosition, null, lightColor, Projectile.rotation, dirtTex.Size() / 2f, Projectile.scale, owner.direction == -1 ? SpriteEffects.FlipVertically : 0f, 0f);
+
+            Main.spriteBatch.Draw(tex, Projectile.Center - Main.screenPosition, null, lightColor, Projectile.rotation, tex.Size() / 2f, Projectile.scale, owner.direction == -1 ? SpriteEffects.FlipVertically : 0f, 0f);
+
+            Color color = Color.Lerp(Color.Transparent, GetRingInsideColor(), shots / (float)MAXSHOTS);
+            //if (reloading)
+                //color = Color.Lerp(GetRingInsideColor(), Color.Transparent, ShootDelay / 90f);
+
+            color.A = 0;
+
+            color *= 0.5f;
+
+            Main.spriteBatch.Draw(glowTex, Projectile.Center - Main.screenPosition, null, color, Projectile.rotation, glowTex.Size() / 2f, Projectile.scale, owner.direction == -1 ? SpriteEffects.FlipVertically : 0f, 0f);
+
+            Main.spriteBatch.Draw(bloomTex, Projectile.Center + (Vector2.One.RotatedBy(Projectile.rotation - MathHelper.PiOver4) * 15f) - Main.screenPosition, null, color, Projectile.rotation, bloomTex.Size() / 2f, 0.95f, owner.direction == -1 ? SpriteEffects.FlipVertically : 0f, 0f);
+            return false;
+        }
+
+        public Color GetRingColor()
+        {
+            switch (SoilType) //switch cant use non constants without this wacko work around
```

Comment:
We should look into a solution for registering the dirt ammo types with their associated data all at once in the same place, instead of placing numerous switch statements around on their type. This is clunky and hurts reuse of the ammo type.